File: Javascript/findDuplicates.js
const findDuplicates = (nums) => {
  const dups = [];
  const trackerObj = {};

  for (let i = 0; i < nums.length; i++) {
    if (trackerObj[nums[i]]) dups.push(nums[i]);
    else trackerObj[nums[i]] = 1;
  }

  return dups;
};

// console.log(findDuplicates([2, 3, 4, 5, 6, 7, 5, 4, 3, 2, 1]));

File: Python/find_duplicates.py
def findDuplicates(nums):
    dups = []
    tracker_dict = {}
    
    for i in range(0, len(nums)):
        try:
            if (tracker_dict[nums[i]] == 1):
                dups.append(nums[i])
        except KeyError:
            tracker_dict[nums[i]] = 1
            
    return dups


# print(findDuplicates([4,3,2,7,8,2,3,1]))
# print(findDuplicates([2,3,4,5,6,7,6,5,4,3,2,1]))